sql/src/main/java/org/apache/druid/sql/calcite/rel/DruidCorrelateUnnestRel.java

```diff
@@ -215,6 +254,26 @@ public DruidQuery toDruidQuery(boolean finalizeAggregations)
     );
   }
 
+  private PartialDruidQuery updateCorrPartialQueryFromLeft(PartialDruidQuery partialQueryFromLeft)
+  {
+    // The DruidCorrelateRule already creates the project and pushes it on the top level
+    // So get select project from partialQuery
+    // The filters are present on the partial query of the left
+    // The group by and having clauses would be on the top level
+    // Same for the sort
+    PartialDruidQuery corrQuery = PartialDruidQuery.create(correlateRel);
+    corrQuery = corrQuery.withWhereFilter(partialQueryFromLeft.getWhereFilter())
+                         .withSelectProject(partialQuery.getSelectProject());
+    if (partialQuery.getAggregate() != null) {
+      corrQuery = corrQuery.withAggregate(partialQuery.getAggregate())
+                           .withHavingFilter(partialQuery.getHavingFilter());
+    }
+    if (partialQuery.getSort() != null || partialQuery.getSortProject() != null) {
+      corrQuery = corrQuery.withSort(partialQuery.getSort());
+    }
+    return corrQuery;
+  }
+
```

Comment:
This feels like we are throwing away the left-hand side and instead trying to build a query from the unnest part?  Taht seems wrong to me, we should be using the unnest on the RHS to augment the query from the LHS (attach to the datasource).  the LHS is the actual working query, the RHS is just the addition of the unnest portion.

Reply:
The DruidCorrelateUnnestRule already put the select project on top of the correlateRel by adding all the things in right to the left. Also the group by, order by on the outer query are already present on the correlateRel. So we create the query by adding the select project (the one which is already projected on top with left + right), the filters from the left + aggregates , sorts if present on the top level